src/components/layermanager/LayermanagerDirective.js

```diff
@@ -323,8 +323,8 @@ goog.require('ga_map_service');
           });
         };
         // Change layers label when topic changes
-        scope.$on('gaLayersChange', function(evt, data) {
-          removeNonTopicLayers(data.topicId);
+        scope.$on('gaTopicChange', function(evt, newTopic) {
+          removeNonTopicLayers(newTopic.id);
```

Comment:
Just something for later: this functionality should probably not be in the layer manager directive...it acts on the map and should be independant of any gui directive that might exist or not.